ToolChain/Assembler/Parser/Parser.py

```diff
@@ -162,239 +179,128 @@ def _parseCodeLine(self, line: list=None):
             possibleCodes = self._findPossibleCodesForInstruction(buildInstruction)
             instructionCode, instructionForm = self._getInstructionCodeAndFormUsingPossibleCodes(buildInstruction,
                                                                                                  possibleCodes)
+
             if instructionForm is None or instructionCode is None:
+                # Code evaluated in function isn't valid
                 raise ValueError("Assembler general error")
+
             else:
+                # Code is valid, fetch the instruction's form based on the arguments, then generate the instruction
                 buildInstruction.instructionCode = instructionCode
                 buildInstruction.instructionLength = formDescription[instructionForm]["length"]
                 lineInstruction = self._generateBinaryInstructionFromInstruction(buildInstruction,
                                                                                  formDescription[instructionForm])
+
         else:
+            # In this case, there were no instructions. Thus, we look at other possibilities
+
             if COMMENT_INDICATORS in buildInstruction.instructionCode:
                 # Simply ignore a comment line
                 buildInstruction = None
                 lineInstruction = None
+
+            #  In these next cases, we can treat them as "data" variables with no real instructions.
+            #  We store the data as the instruction's source immediate value, and set instruction code to 0
+
             elif DATA_ALPHA_INDICATOR in buildInstruction.instructionCode:
-                # This is a bunch of alpha data
+                # This is a bunch of alpha data (String)
                 lineInstruction = buildInstruction.sourceImmediate
-                buildInstruction.instructionCode = 0  # Calling code expect this to be non STR for non mem ref
+                buildInstruction.instructionCode = 0
+
             elif DATA_NUMERIC_INDICATOR in buildInstruction.instructionCode:
-                # This is a numeric data field
+                # This is a numeric data field (Integer)
                 lineInstruction = struct.pack(">I", buildInstruction.sourceImmediate)
-                buildInstruction.instructionCode = 0  # Calling code expect this to be non STR for non mem ref
+                buildInstruction.instructionCode = 0
+
             elif DATA_MEMORY_REFERENCE in buildInstruction.instructionCode:
+                # This is a data memory reference (dataMemref)
                 lineInstruction = buildInstruction.sourceImmediate
                 buildInstruction.instructionCode = 0
+
             else:
+                # Having exhausted all options, we assume there was nothing else on this line.
                 lineInstruction = None
 
         return buildInstruction, lineInstruction
 
-    def translateRegisterNameToRegisterCode(self, registerName: str=""):
```

Comment:
Why did you move this?

Reply:
Yeah, provided an explanation for moving it. Could be alphabetical or order of appearance, but make sure everything else matches that scheme.

Reply:
The idea was to have more "utility" methods at the end, apart from the main functions. Though this could be subjective, perhaps there is a better way to organize the methods? Alphabetical could work as well.